openbsd-compat/getrrsetbyname.c

```diff
@@ -123,6 +123,14 @@ _getshort(msgp)
 	GETSHORT(u, msgp);
 	return (u);
 }
+#elif HAVE__GETSHORT == 2
```

Comment:
I think you could do away with the magic value with:

#if defined(HAVE_DECL__GETSHORT) && HAVE_DECL__GETSHORT == 1 && !defined(HAVE__GETSHORT)

Reply:
Don't get me started on this one. I am opening a case with the HPE compiler group because there is something seriously wrong here. The issue is that we have `_getshort()` declared in `/usr/include/resolv.h`, but it appears not to be exported anywhere (I have searched every DLL on both platform variants. If I don't configure HAVE_GETSHORT, I get into a conflicting declaration.

Reply:
it conflicts because the function is const-ified and the declaration is not?  The #define const hack might solve that?

Reply:
It conflicts, but there is no actual implementation for `_getshort` or `_getlong` in the operating system even though it is declared. `/usr/include/resolv.h` is `#included` through a platform header file so these declarations interfere with declaring our own. The existing implementation in `openbsd-compat/getrrsetbyname.c` conflicts with the compiler's POV of what is in `resolv.h`, so we have the DECL. I'm thinking we force #undef HAVE_GETSHORT in config.h and do it the way @daztucker suggests. If would be nice to tell configure to do that for us (suggestions?).

Reply:
actually I think the first stop would be to see if we can remove the "register" upstream since it's obsolescent anyway.

Reply:
and in fact it looks like these definitions came from a different file and were de-registered a long time ago:
https://cvsweb.openbsd.org/cgi-bin/cvsweb/src/lib/libc/net/res_comp.c#rev1.12

Reply:
> and in fact it looks like these definitions came from a different file and were de-registered a long time ago: https://cvsweb.openbsd.org/cgi-bin/cvsweb/src/lib/libc/net/res_comp.c#rev1.12

Should I leave things alone for now? But how did they get here in that case?

Reply:
If it makes things better for you I'd delete them in your tree.  I'll remove them from our tree after the 8.9 release (it's *probably* a low-impact change, but it doesn't fix anything that previously worked, so it can wait).

They aren't in the revision of gerrsetbyname.c that was imported, so it looks like Damien added them to make the port work:
https://github.com/openssh/openssh-portable/commit/d9ec370ac3a27fc9374c4182b7477ec0fc8473f4

Reply:
> If it makes things better for you I'd delete them in your tree. I'll remove them from our tree after the 8.9 release (it's _probably_ a low-impact change, but it doesn't fix anything that previously worked, so it can wait).
> 
> They aren't in the revision of gerrsetbyname.c that was imported, so it looks like Damien added them to make the port work: [d9ec370](https://github.com/openssh/openssh-portable/commit/d9ec370ac3a27fc9374c4182b7477ec0fc8473f4)

Will do... wait... doesn't d9ec370 that violate rule 4? ;)

Reply:
(for context, that was "make the minimum changes to files that we keep in sync with OpenBSD").

Maybe?  On one hand putting _getshort and friends in their own file would make it both files easier to sync, but on the other had you then can't make them static so have potential symbol pollution and higher chance of header collisions similar to yours.

Anyway that commit was nearly 20 years ago, it's possible we've gained some experience about what works for us since then.

Reply:
I don't know how `#if defined(HAVE_DECL__GETSHORT) && HAVE_DECL__GETSHORT == 1 && !defined(HAVE__GETSHORT)` works, because the prior block defines the incompatible `_getshort` before this `#if` gets invoked. Putting it first changes the semantics of the if. I'm a bit stuck on this one.

Reply:
I've just committed the register removal, might be worth rebasing.